lesson17/src/js/main.js

```diff
@@ -0,0 +1,212 @@
+function showLoadingImage() {
+  const main = document.getElementById("main");
+  const fragment = document.createDocumentFragment();
+  const renderLoadingBox = document.createElement("div");
+  renderLoadingBox.classList.add("loading");
+  renderLoadingBox.id = "js-loading";
+  const loadingImage = document.createElement("img");
+  loadingImage.src = "images/loading.gif";
+  loadingImage.width = 80;
+  loadingImage.height = 80;
+  loadingImage.classList.add("loading__image");
+  fragment.appendChild(renderLoadingBox).appendChild(loadingImage);
+  document.body.insertBefore(fragment, main);
+}
+
+function removeLoading() {
+  document.getElementById("js-loading").remove();
+}
+
+function displayErrorMassage(error) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+  const textBox = document.createElement("p");
+  textBox.classList.add("");
+  textBox.textContent = error;
+  slideImageArea.appendChild(textBox);
+}
+
+function renderSlideArea() {
+  const slideImageArea = document.createElement("div");
+  slideImageArea.classList.add("mainvisual__images__inner");
+
+  document.querySelector(".mainvisual__images").appendChild(slideImageArea);
+}
+
+function renderSlideImage(item) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+
+  const slider = document.createElement("ul");
+  slider.classList.add("slider");
+
+  const fragment = document.createDocumentFragment();
+  item.forEach((image, i) => {
+    const sliderItem = document.createElement("li");
+    sliderItem.classList.add("slider__item");
+    sliderItem.setAttribute("data-slide-index", i + 1);
```

Comment:
IMO
Why not use dataset.index?
I suggested changing the function after this, so I removed the '+1'.
```suggestion
    sliderItem.dataset.index = i;
```

Reply:
I didn't know the properties of the dataset.
I learned this!

[467e823](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/467e823afb67e8aefedf7a072d3219533fa0ca9a)